numba/parfors/parfor_lowering.py

```diff
@@ -1621,16 +1621,18 @@ def load_potential_tuple_var(x):
                     array_strides.append(strides[j])
                 builder.store(builder.bitcast(ary.data, byte_ptr_t), dst)
         else:
+            # do not allow any other subclasses
+            aty_is_bool = type(aty) in (types.Boolean, types.BooleanLiteral)
```

Comment:
`types.BooleanLiteral` inherits from `types.Boolean`, could this just be an `isinstance` check like `aty_is_bool = isinstance(aty, types.Boolean)` ?

Reply:
I note this is discussed here https://github.com/numba/numba/pull/6015#discussion_r460274711, the current restrictive check does not follow what has been used elsewhere in Numba (or parfors), are there known use cases where this is a problem? Also Numba types have an `is_internal` property to determine if a type is part of the Numba type system or a user defined type, would suggest using that if restrictive behaviour is required.

CC @DrTodd13 any view on the above?

Reply:
@stuartarchibald Yes, thank you, using the following definition works too.
`aty_is_bool = isinstance(aty, types.Boolean) and aty.is_internal`

I'm not aware of any issues related to user sub-classing from types.Boolean and affecting parfor lowering, and I agree that changing it in one place only is probably not a good idea. So I can leave it as simple  `isinstance(aty, types.Boolean)` or use above definition. Just need somebody to tell which one to use.

Reply:
@kozlov-alexey,  @DrTodd13 and I just had an extensive out of band discussion about this and have concluded that a less restrictive `isinstance(aty, types.Boolean)` type check would be good for now. The implications of sub-classing from core Numba types in a breaking manner are not considered elsewhere in the code base and were such a case to occur there's a few options for resolving this, but such a solutions are inevitably quite wide reaching and would impact more than `parfors`. Hows that sound? Thanks!

Reply:
@kozlov-alexey think it's just ^ to resolve then this is good to merge. Thanks.